src/main/scala/ai/privado/languageEngine/python/passes/config/PythonPropertyFilePass.scala

```diff
@@ -0,0 +1,179 @@
+package ai.privado.languageEngine.python.passes.config
+
+import ai.privado.utility.Utilities
+import io.joern.x2cpg.SourceFiles
+import io.shiftleft.codepropertygraph.generated.{Cpg, EdgeTypes}
+import io.shiftleft.codepropertygraph.generated.nodes.{Literal, Member, NewFile, NewJavaProperty}
+import io.shiftleft.passes.ForkJoinParallelCpgPass
+import org.slf4j.LoggerFactory
+import overflowdb.BatchedUpdate
+
+import scala.jdk.CollectionConverters._
+import java.util.Properties
+import scala.util.{Failure, Success, Try}
+import io.shiftleft.semanticcpg.language._
+import com.typesafe.config._
+
+import java.io.File
+import scala.io.Source
+
+class PythonPropertyFilePass(cpg: Cpg, projectRoot: String) extends ForkJoinParallelCpgPass[String](cpg) {
+  override def generateParts(): Array[String] = {
+    configFiles(projectRoot, Set(".ini", ".yml", ".yaml", ".env")).toArray
+  }
+
+  private val logger = LoggerFactory.getLogger(getClass)
+
+  override def runOnPart(builder: DiffGraphBuilder, file: String): Unit = {
+    val fileNode      = addFileNode(file, builder)
+    val propertyNodes = addPropertyNodesAndConnectToUsers(file, builder)
+    propertyNodes.foreach(builder.addEdge(_, fileNode, EdgeTypes.SOURCE_FILE))
+  }
+
+  private def addPropertyNodesAndConnectToUsers(
+    file: String,
+    builder: BatchedUpdate.DiffGraphBuilder
+  ): List[NewJavaProperty] = {
+    Try {
+      obtainKeyValuePairs(file).filter(pair => pair._1.nonEmpty && pair._2.nonEmpty)
+    } match {
+      case Success(keyValuePairs) =>
+        val propertyNodes = keyValuePairs.map(addPropertyNode(_, builder))
+
+        propertyNodes.foreach(propertyNode => {
+          connectGetEnvironLiterals(propertyNode, builder)
+          connectDBConfigMembers(propertyNode, builder)
+        })
+
+        propertyNodes
+      case Failure(exception) =>
+        logger.warn(exception.getMessage)
+        List()
+    }
+  }
+
+  private def obtainKeyValuePairs(file: String): List[(String, String)] = {
+    if (file.endsWith(".conf")) {
+      parseConfFiles(file)
+    } else if (file.endsWith(".ini")) {
+      parseINIFiles(file)
+    } else {
+      getDotenvKeyValuePairs(file)
+    }
+  }
+
+  private def getDotenvKeyValuePairs(file: String): List[(String, String)] = {
```

Comment:
write test case for this function

Reply:
No explicit test case is required as this run as part of all .env files. Test cases are written as subpart of .env file tests.